lib/diagnostic.js

```diff
@@ -6,26 +6,42 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (words) {
+  return words.replace(/\W+/,'').toUpperCase();
+  }
 };
 
+let normalizedWords = words.map(normalize);
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (array) {
+  let result = {};
+  array.forEach (function (words) {
+    result[words] = null;
+  });
+  return result;
 };
 
```

Comment:
This is really good, but balks because words is undefined on line 14, so you aren't passing an array into uniqueWords.